### ISSUE TITLE
Websocket compression handshake was blocked by Application Gateway

### ISSUE THREAD

**Issue by deshengio** [Azure/service-fabric-issues#903]
besides the issue described in (https://github.com/Azure/service-fabric-issues/issues/421) , another issue related to "Sec-WebSocket-Extensions" is:
Websocket request with compression feature will have header like:
```
Max-Frame-Size: Sec-WebSocket-Extensions: permessage-deflate
8192
```

this kind of request will be rejected by application gateway with error 400:
```
HTTP/1.1 400 Bad Request
Content-Type: text/html; charset=us-ascii
Server: Microsoft-HTTPAPI/2.0
Date: Tue, 06 Mar 2018 15:12:00 GMT
Connection: close
Content-Length: 339

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Bad Request</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY><h2>Bad Request - Invalid Header</h2>
<hr><p>HTTP Error 400. The request has an invalid header name.</p>
</BODY></HTML>
```

However, same application works well when Azure basic load balancer is used.
